doc/specs/stdlib_hash_procedures.md

```diff
@@ -0,0 +1,1748 @@
+E. Knuth. It multiplies the `KEY` by the odd valued approximation to
+`2**64/phi`, where `phi` is the golden ratio 1.618..., and returns the
+`nbits` upper bits of the product as the lowest bits of the result.
+
+##### Example
+
+```fortran
+    program demo_fibonacci_hash
+      use stdlib_64_bit_hash_codes, only: fibonacci_hash
+      use iso_fortran_env, only: int64 
+      implicit none
+      integer, allocatable :: array1(:)
+      integer(int64) :: hash, source
+      allocate( array1(0:2**6-1) )
+      array1(:) = 0
+      source = int(Z'1FFFFFFFF', int64)
+      hash = fibonacci_hash(source, 6)
+      azray1(hash) = source
+      print *, hash
+    end program demo_fibonacci_hash
+```
+
+#### `FNV_1`- calculates a hash code from a key
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 64 bit hash code from a rank 1 integer array or a default
+character string.
+
+##### Syntax
+
+`code = [[stdlib_64_bit_hash_codes:fnv_1]]( key )`
+
+##### Class
+
+Pure/elemental function
+
+##### Argument
+
+`key`: Shall be a deferred length default character scalar expression
+or a rank 1 integer array expression of kind `INT8`, `INT16`,
+`INT32`, or `INT64`.
+It is an `intent(in)` argument.
+
+##### Result
+
+The result is a scalar integer of kind `INT64`. 
+
+##### Note
+
+`FNV_1` is an implementation of the original FNV-1 hash code of Glenn
+Fowler, Landon Curt Noll, and Phong Vo.
+It differs from typical implementations in that it also ecodes the
+size of the structure in the hash code.
+This code is relatively fast on short keys, and is small enough that it
+will often be retained in the instruction cache if hashing is
+intermittent.
+As a result it should give good performance for typical hash table
+applications, although it is rare for them to need 64 bits.
+This code does not pass any of the SMHasher tests, but the resulting
+degradation in performance due to its larger number of collisions is
+expected to be minor compared to its faster hashing rate.
+It is a *pure* function for integer arrays, and an *elemental*
+function for character strings. 
+
+
+##### Example
+
+```fortran
+    program demo_fnv_1_hash
+      use stdlib_64_bit_hash_codes, only: fnv_1_hash
+	  use iso_fortran_env, only: int64
+      implicit none
+      integer, allocatable :: array1(:)
+      integer(int64) :: hash
+      array1 = [ 5, 4, 3, 1, 10, 4, 9]
+      hash = fnv_1_hash(array1)
+      print *, hash
+    end program demo_fnv_1_hash
+```
+
+
+#### `FNV_1A`- calculates a hash code from a key
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 64 bit hash code from a rank 1 integer array or a default
+character string.
+
+##### Syntax
+
+`code = [[stdlib_64_bit_hash_codes:fnv_1a]]( key )`
+
+##### Class
+
+Pure/elemental function
+
+##### Argument
+
+`key`: Shall be a deferred length default character scalar expression
+or a rank 1 integer array expression of kind `INT8`, `INT16`,
```

Comment:
```suggestion
or a rank-1 integer array expression of kind `INT8`, `INT16`,
```

Not sure if specific rank references should also be hyphenated? Perhaps best to consult MRC or the standard document and see what they use.

Reply:
They have been hyphenated.